src/neural_de.jl

```diff
@@ -167,3 +167,29 @@ function (n::NeuralCDDE)(x,p=n.p)
     prob = DDEProblem{false}(dudt_,x,n.hist,n.tspan,p,constant_lags = n.lags)
     concrete_solve(prob,n.solver,x,p,n.args...;sensealg=TrackerAdjoint(),n.kwargs...)
 end
+
+struct NeuralDAE{P,M,RE,T,S,DV,A,K} <: NeuralDELayer
```

Comment:
This does not cover the mass matrix form. That should be added as per https://docs.juliadiffeq.org/latest/tutorials/advanced_ode_example/#Handling-Mass-Matrices-1 and the discussion in the slack thread.